ros2lifecycle_test_fixtures/src/simple_lifecycle_node.cpp

```diff
@@ -0,0 +1,94 @@
+// Copyright 2019 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <chrono>
+#include <iostream>
+#include <memory>
+#include <string>
+#include <thread>
+#include <utility>
+
+#include "rclcpp/rclcpp.hpp"
+
+#include "rclcpp_lifecycle/lifecycle_node.hpp"
+
+using namespace std::chrono_literals;
+
+class SimpleLifecycleNode : public rclcpp_lifecycle::LifecycleNode
+{
+public:
+  explicit SimpleLifecycleNode(const std::string & node_name)
+  : rclcpp_lifecycle::LifecycleNode(node_name)
+  {}
+
+  rclcpp_lifecycle::node_interfaces::LifecycleNodeInterface::CallbackReturn
+  on_configure(const rclcpp_lifecycle::State &)
+  {
+    RCLCPP_INFO(get_logger(), "on_configure() is called.");
+
+    return rclcpp_lifecycle::node_interfaces::LifecycleNodeInterface::CallbackReturn::SUCCESS;
+  }
+
+  rclcpp_lifecycle::node_interfaces::LifecycleNodeInterface::CallbackReturn
+  on_activate(const rclcpp_lifecycle::State &)
+  {
+    RCLCPP_INFO(get_logger(), "on_activate() is called.");
+
+    return rclcpp_lifecycle::node_interfaces::LifecycleNodeInterface::CallbackReturn::SUCCESS;
+  }
+
+  rclcpp_lifecycle::node_interfaces::LifecycleNodeInterface::CallbackReturn
+  on_deactivate(const rclcpp_lifecycle::State &)
+  {
+    RCLCPP_INFO(get_logger(), "on_deactivate() is called.");
+
+    return rclcpp_lifecycle::node_interfaces::LifecycleNodeInterface::CallbackReturn::SUCCESS;
+  }
+
+  rclcpp_lifecycle::node_interfaces::LifecycleNodeInterface::CallbackReturn
+  on_cleanup(const rclcpp_lifecycle::State &)
+  {
+    RCLCPP_INFO(get_logger(), "on cleanup is called.");
+
+    return rclcpp_lifecycle::node_interfaces::LifecycleNodeInterface::CallbackReturn::SUCCESS;
+  }
+
+  rclcpp_lifecycle::node_interfaces::LifecycleNodeInterface::CallbackReturn
+  on_shutdown(const rclcpp_lifecycle::State & state)
+  {
+    RCLCPP_INFO(
+      get_logger(),
+      "on shutdown is called from state %s.",
+      state.label().c_str());
+
+    return rclcpp_lifecycle::node_interfaces::LifecycleNodeInterface::CallbackReturn::SUCCESS;
+  }
+};
+
+int main(int argc, char * argv[])
+{
+  // force flush of the stdout buffer.
+  // this ensures a correct sync of all prints
+  setvbuf(stdout, NULL, _IONBF, BUFSIZ);
+
+  rclcpp::init(argc, argv);
+
+  auto lc_node = std::make_shared<SimpleLifecycleNode>("lc_node");
```

Comment:
Might be worth renaming the node to something more specific (and different from the lifecycle talker exec in the `lifecycle` demo package).

Reply:
I use `"simple_lc_node"` on `launch`, does that sound reasonable? Or would you rather be more specific?

Reply:
that's okay I guess. I personally like to prefix them with `test` or the name of the test package in this case, but either way is fine.

Reply:
See https://github.com/ros2/ros2cli/pull/391/commits/046d82c2903692af9c29d055c084de85422a9393